src/parser/expression_parser.ml

```diff
@@ -860,6 +860,43 @@ module Expression
     Expect.token env (T_NUMBER { kind; raw });
     value
 
+  (* TODO: don't know if this function should be here *)
```

Comment:
Yeah, it's a little non-ideal that this is duplicated between here and the lexer. That said, `Flow_lexer.mk_num_singleton` and `Expression_parser.number` seem awfully similar. So what you're doing is no worse than what already exists. And the code is slightly different, I suppose :P